includes/class.llms.cache.helper.php

```diff
@@ -31,6 +31,30 @@ public function __construct() {
 
 	}
 
+	/**
+	 * Sets a browser cookie that tells WP Engine to exclude a page from server caching.
+	 *
+	 * If "Settings -> Permalinks" is "Plain", i.e. the `permalink_structure` option is '',
+	 * then the entire site will be excluded from server caching.
```

Comment:
I didn't notice this on my first read through and I understand why it's here (because we can't set a cookie based on query string vars).

I wondering if this is an acceptable condition for us to place on a WPE site with plain permalinks.

I'm considering the "safer" option might be to only add the cookie when there is a permalink structure.

Very few sites actually use plain permalinks (on purpose) but it still feels a bit arrogant to say that LifterLMS disables caching if you use plain permalinks on WPE...

The only other alternative I could consider would be to set the cookie and then also unset the cookie for any other page load. But that's kind of crazy I think...

Thoughts @gocodebox/engineering

Reply:
I think that sooner or later we should stop allowing ugly permalinks, there are already parts of our code which rely on pretty permalinks, e.g. our rest api. Also there are other plugins that require pretty permalinks, e.g. BuddyPress.

That said I'm with avoid setting the cookie when using ugly permalinks.

Reply:
If the WordPress permalink structure is set to plain, e.g. the LifterLMS dashboard is `https://example.com/?p=17`, our options are:
1. Disable WP Engine server side caching on the `/` path, which means the whole site, unless some other plugins use their own permalink structure.
2. Don't disable the cache, which results in an "The password reset key could not be found. Please reset your password again if needed." error displayed to the user.
3. Don't disable the cache. Add an admin notice explaining the conflict with WP Engine server-side caching, the plain permalink structure, and the reset password feature. Their options are to change the permalink structure, configure exclusion paths, or deal with unhappy users.
4. Don't disable the cache and disable the lost password feature.
5. Don't disable the cache and change the error message in [`llms_parse_password_reset_cookie()`](https://github.com/gocodebox/lifterlms/blob/6.4.0/includes/functions/llms.functions.person.php#L360) if the cookie is not found, the permalink structure is plain and `is_wpe()` exists and returns 1.
6. Bust the cache with an extra query variable, like `https://example.com/?p=17&llms_time=1651088529`. I remember experimenting with this technique, but forgot why I didn't choose it.

Reply:
@thomasplevy, which option would you like me to implement?

Reply:
I think 2 and then create a piece of documentation in the knowledge base detailing this as a "known issue" on WPE. As long as we know about it and have documentation outlining the reasons why we arrived at this option (summarizing this conversation) I think that's best.

Adding custom error messages or additional custom handlers specifically for this double-niche scenario (on WPE & using plain permalinks) seems unnecessary.

This function should chec `is_wpe()` and whether or not permalinks are enabled. That seems good to me.

Reply:
@willmiddleton-lifterlms, a rough draft of a KB post about this issue in [here](https://lifterlms.com/?post_type=pp-doc&p=210802&preview=true). Please give it your magic touch.